shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java

```diff
@@ -113,6 +113,9 @@ public SelectStatementContext(final Map<String, ShardingSphereMetaData> metaData
     private ShardingSphereSchema getSchema(final Map<String, ShardingSphereMetaData> metaDataMap, final String defaultSchemaName) {
         String schemaName = tablesContext.getSchemaName().orElse(defaultSchemaName);
         ShardingSphereMetaData metaData = metaDataMap.get(schemaName);
+        if (null == metaData && tablesContext.getTables().isEmpty()) {
+            return null;
+        }
         Preconditions.checkState(null != metaData, "Can not get meta data by schema mame `%s`.", schemaName);
```

Comment:
Unnecessary code

Reply:
> Unnecessary code？

@menghaoranss For sql without table, like `select 1`, it doesn't need metaData.